<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="William Howard Beasley (Howard Live Oak LLC, Norman)" />
<meta name="author" content="Joseph Lee Rodgers (Vanderbilt University, Nashville)" />
<meta name="author" content="David Bard (University of Oklahoma Health Sciences Center, OKC)" />
<meta name="author" content="Kelly Williams (Oklahoma City University, OKC)" />
<meta name="author" content="Michael D. Hunter (Penn State)" />

<meta name="date" content="2024-10-07" />

<title>ACE Models with the NLSY</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">ACE Models with the NLSY</h1>
<h4 class="author">William Howard <a href="https://scholar.google.com/citations?user=ffsJTC0AAAAJ">Beasley</a>
(Howard Live Oak LLC, Norman)</h4>
<h4 class="author">Joseph Lee <a href="https://www.vanderbilt.edu/psychological_sciences/bio/joe-rodgers">Rodgers</a>
(Vanderbilt University, Nashville)</h4>
<h4 class="author">David <a href="https://medicine.ouhsc.edu/Academic-Departments/Pediatrics/Sections/Developmental-Behavioral-Pediatrics/Faculty/david-e-bard-phd">Bard</a>
(University of Oklahoma Health Sciences Center, OKC)</h4>
<h4 class="author">Kelly <a href="https://oklahoma.gov/omma.html">Williams</a> (Oklahoma City
University, OKC)</h4>
<h4 class="author">Michael D. <a href="https://acquia-prod.hhd.psu.edu/contact/michael-hunter">Hunter</a>
(Penn State)</h4>
<h4 class="date">2024-10-07</h4>


<div id="TOC">
<ul>
<li><a href="#terminology" id="toc-terminology">Terminology</a></li>
<li><a href="#example-df-analysis-with-a-simple-outcome-for-gen2-subjects-using-a-package-variable" id="toc-example-df-analysis-with-a-simple-outcome-for-gen2-subjects-using-a-package-variable">Example:
DF analysis with a Simple Outcome for Gen2 Subjects, Using a Package
Variable</a></li>
<li><a href="#example-df-analysis-with-a-univariate-outcome-from-a-gen2-extract" id="toc-example-df-analysis-with-a-univariate-outcome-from-a-gen2-extract">Example:
DF analysis with a univariate outcome from a Gen2 Extract</a></li>
<li><a href="#example-multiple-group-sem-of-a-simple-outcome-for-gen2-subjects" id="toc-example-multiple-group-sem-of-a-simple-outcome-for-gen2-subjects">Example:
Multiple Group SEM of a Simple Outcome for Gen2 Subjects</a></li>
<li><a href="#example-multiple-group-sem-of-a-simple-outcome-for-gen1-subjects" id="toc-example-multiple-group-sem-of-a-simple-outcome-for-gen1-subjects">Example:
Multiple Group SEM of a Simple Outcome for Gen1 Subjects</a></li>
<li><a href="#example-multiple-group-sem-of-a-simple-outcome-all-pairs-in-gen1-and-gen2" id="toc-example-multiple-group-sem-of-a-simple-outcome-all-pairs-in-gen1-and-gen2">Example:
Multiple Group SEM of a Simple Outcome all pairs in Gen1 and
Gen2</a></li>
<li><a href="#appendix-receiving-help-for-the-nlsylinks-package" id="toc-appendix-receiving-help-for-the-nlsylinks-package">Appendix:
Receiving Help for the NlsyLinks Package</a></li>
<li><a href="#appendix-creating-and-saving-r-scripts" id="toc-appendix-creating-and-saving-r-scripts">Appendix: Creating and
Saving R Scripts</a></li>
<li><a href="#appendix-installing-and-loading-the-nlsylinks-package" id="toc-appendix-installing-and-loading-the-nlsylinks-package">Appendix:
Installing and Loading the NlsyLinks Package</a></li>
<li><a href="#appendix-references" id="toc-appendix-references">Appendix: References</a></li>
<li><a href="#funding" id="toc-funding">Funding</a></li>
<li><a href="#version-information" id="toc-version-information">Version
Information</a></li>
</ul>
</div>

<p>We describe how to use the NlsyLinks package to examine various
biometric models, using the NLSY79.</p>
<div id="terminology" class="section level1">
<h1>Terminology</h1>
<p>Researchers and grad students interested using the <a href="https://www.nlsinfo.org/">NLSY</a> for Behavior Genetics and
family research, please start with our 2016 article, <a href="https://link.springer.com/article/10.1007/s10519-016-9785-3"><em>The
NLSY Kinship Links: Using the NLSY79 and NLSY-Children Data to Conduct
Genetically-Informed and Family-Oriented Research</em></a>.</p>
<p>This package considers both Gen1 and Gen2 subjects. ‘Gen1’ refers to
subjects in the original NLSY79 sample (<a href="https://www.nlsinfo.org/content/cohorts/nlsy79" class="uri">https://www.nlsinfo.org/content/cohorts/nlsy79</a>). ‘Gen2’
subjects are the biological offspring of the Gen1 females
-<em>i.e.</em>, those in the NLSY79 Children and Young Adults sample (<a href="https://www.nlsinfo.org/content/cohorts/nlsy79-children" class="uri">https://www.nlsinfo.org/content/cohorts/nlsy79-children</a>).
The NLSY97 is a third dataset that can be used for behavior genetic
research (<a href="https://www.nlsinfo.org/content/cohorts/nlsy97" class="uri">https://www.nlsinfo.org/content/cohorts/nlsy97</a>),
although this vignette focuses on the two generations in the NLSY79.</p>
<p>Standard terminology is to refer second generation subjects as
‘children’ when they are younger than age 15 (NSLYC), and as ‘young
adults’ when they are 15 and older (NLSY79-YA); though they are the same
respondents, different funding mechanisms and different survey items
necessitate the distinction. This cohort is sometimes abbreviated as
‘NLSY79-C’, ‘NLSY79C’, ‘NLSY-C’ or ‘NLSYC’. This packages uses ‘Gen2’ to
refer to subjects of this generation, regardless of their age at the
time of the survey.</p>
<p>Within our own team, we’ve mostly stopped using terms like ‘NLSY79’,
‘NLSY79-C’ and ‘NLSY79-YA’, because we conceptualize it as one big
sample containing two related generations. It many senses, the responses
collected from the second generation can be viewed as outcomes of the
first generation. Likewise, the parents in the first generation provide
many responses that can be viewed as explanatory variables for the 2nd
generation. Depending on your research, there can be big advantages of
using one cohort to augment the other. There are also survey items that
provide information about the 3rd generation and the 0th generation.</p>
<p>The <code>SubjectTag</code> variable uniquely identify NLSY79
subjects when a dataset contains both generations. For Gen2 subjects,
the <code>SubjectTag</code> is identical to their CID (<em>i.e.</em>,
C00001.00 -the ID assigned in the NLSY79-Children files). However for
Gen1 subjects, the <code>SubjectTag</code> is their CaseID
(<em>i.e.</em>, R00001.00), with “00” appended. This manipulation is
necessary to identify subjects uniquely in inter-generational datasets.
A Gen1 subject with an ID of 43 becomes 4300. The
<code>SubjectTag</code>s of her four children remain 4301, 4302, 4303,
and 4304.</p>
<p>The <em>expected coefficient of relatedness</em> of a pair of
subjects is typically represented by the statistical variable
<code>R</code>. Examples are: Monozygotic twins have <code>R</code>=1;
dizygotic twins have <code>R</code>=0.5; full siblings (<em>i.e.</em>,
those who share both biological parents) have <code>R</code>=0.5;
half-siblings (<em>i.e.</em>, those who share exactly one biological
parent) have <code>R</code>=0.25; adopted siblings have
<code>R</code>=0.0. Other uncommon possibilities are mentioned the
documentation for <code>Links79Pair</code>. The font (and hopefully
their context) should distinguish the variable <code>R</code> from the
software R. To make things slightly more confusing the computer variable
for <code>R</code> in the <code>Links79Pair</code> dataset is written
with a monospace font: <code>R</code>.</p>
<p>A subject’s <code>ExtendedID</code> indicates their extended family.
Two subjects will be in the same extended family if either: [1] they are
Gen1 housemates, [2] they are Gen2 siblings, [3] they are Gen2 cousins
(<em>i.e.</em>, they have mothers who are Gen1 sisters in the NLSY79),
[4] they are mother and child (in Gen1 and Gen2, respectively), or [5]
they are (aunt|uncle) and (niece|nephew) (in Gen1 and Gen2,
respectively).</p>
<p>An <strong>outcome variable</strong> is directly relevant to the
applied researcher; these might represent constructs like height, IQ,
and income. A <strong>plumbing variable</strong> is necessary to manage
BG datasets; examples are <code>R</code>, a subject’s ID, and the date
of a subject’s last survey.</p>
<p>An ACE model is the basic biometrical model used by Behavior Genetic
researchers, where the genetic and environmental effects are assumed to
be additive. The three primary variance components are (1) the
proportion of variability due to a shared genetic influence (typically
represented as <span class="math inline">\(a^2\)</span>, or sometimes
<span class="math inline">\(h^2\)</span>), (2) the proportion of
variability due to shared common environmental influence (typically
<span class="math inline">\(c^2\)</span>), and (3) the proportion of
variability due to unexplained/residual/error influence (typically <span class="math inline">\(e^2\)</span>).</p>
<p>The variables are scaled so that they account for all observed
variability in the outcome variable; specifically: <span class="math inline">\(a^2 + c^2 + e^2 = 1\)</span>. Using appropriate
designs that can logically distinguish these different components (under
carefully specified assumptions), the basic biometrical modeling
strategy is to estimate the magnitude of <span class="math inline">\(a^2\)</span>, <span class="math inline">\(c^2\)</span>, and <span class="math inline">\(e^2\)</span> within the context of a particular
model. For gentle introductions to Behavior Genetic research, we
recommend <a href="https://books.google.com/books?id=r7AgAQAAIAAJ&amp;source=gbs_navlinks_s">Plomin
(1990)</a> and <a href="https://www.colorado.edu/psych-neuro/gregory-carey">Carey
(2003)</a>. For more in-depth ACE model-fitting strategies, we recommend
<a href="http://ibgwww.colorado.edu/workshop2006/cdrom/HTML/book2004a.pdf">Neale
&amp; Maes, (2004)</a>.</p>
<p>The <strong>NLS Investigator</strong> (<a href="https://www.nlsinfo.org/investigator/" class="uri">https://www.nlsinfo.org/investigator/</a>) is the best way
to obtain the NLSY79 and NLSY97 datasets. See our vignette dedicated to
the NLS Investigator by typing <code>vignette(&quot;NlsInvestigator&quot;)</code>
or by visiting <a href="https://cran.r-project.org/package=NlsyLinks" class="uri">https://cran.r-project.org/package=NlsyLinks</a>.</p>
<p>Before starting the real examples, first verify that the NlsyLinks
package is installed correctly. If not, please refer to Appendix.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">any</span>(<span class="fu">.packages</span>(<span class="at">all.available =</span> <span class="cn">TRUE</span>) <span class="sc">==</span> <span class="st">&quot;NlsyLinks&quot;</span>) <span class="co"># Should evaluate to TRUE.</span></span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">library</span>(NlsyLinks) <span class="co"># Load the package into the current session.</span></span></code></pre></div>
<p>The package’s documentation manual can be opened by typing
<code>?NlsyLinks</code> in R or clicking the appropriate entry in
RStudio’s ‘Packages’ tab (which is usually in the lower right
panel).</p>
</div>
<div id="example-df-analysis-with-a-simple-outcome-for-gen2-subjects-using-a-package-variable" class="section level1">
<h1>Example: DF analysis with a Simple Outcome for Gen2 Subjects, Using
a Package Variable</h1>
<p>The vignette’s first example uses a simple statistical model and all
available Gen2 subjects. The <code>CreatePairLinksDoubleEntered</code>
function will create a data frame where each represents one pair of
siblings, respective of order (<em>i.e.</em>, there is a row for
Subjects 201 and 202, and a second row for Subjects 202 and 201). This
function examines the subjects’ IDs and determines who is related to
whom (and by how much). By default, each row it produces has at least
six values/columns: (i) ID for the older member of the kinship pair:
<code>Subject1Tag</code>, (ii) ID for the younger member:
<code>Subject2Tag</code>, (iii) ID for their extended family:
<code>ExtendedID</code>, (iv) their estimated coefficient of genetic
relatedness: <code>R</code>, (v <em>and beyond</em>) outcome values for
the older member; (vi <em>and beyond</em>) outcome values for the
younger member.</p>
<p>A DeFries-Fulker (<strong>DF</strong>) Analysis uses linear
regression to estimate the <span class="math inline">\(a^2\)</span>,
<span class="math inline">\(c^2\)</span>, and <span class="math inline">\(e^2\)</span> of a univariate biometric system. The
interpretations of the DF analysis can be found in Rodgers &amp; Kohler
(2005) and Rodgers, Rowe, &amp; Li (1999). This vignette example uses
the newest variation, which estimates two parameters; the corresponding
function is called <code>DeFriesFulkerMethod3</code>. The steps are:</p>
<ol style="list-style-type: decimal">
<li><p>Use the NLS Investigator to select and download a Gen2
dataset.</p></li>
<li><p>Open R and create a new script (see <a href="#appendix-creating-and-saving-r-scripts">Appendix: R Scripts</a>
and load the NlsyLinks package. If you haven’t done so, <a href="#appendix-installing-and-loading-the-nlsylinks-package">install
the NlsyLinks package</a>). Within the R script, identify the locations
of the downloaded data file, and load it into a data frame.</p></li>
<li><p>Within the R script, load the linking dataset. Then select only
Gen2 subjects. The ‘Pair’ version of the linking dataset is essentially
an upper triangle of a symmetric sparse matrix.</p></li>
<li><p>Load and assign the <code>ExtraOutcomes79</code>
dataset.</p></li>
<li><p>Specify the outcome variable name and filter out all subjects who
have a negative value in this variable. The NLSY typically uses negative
values to indicate different types of missingness (see ‘Further
Information’ below).</p></li>
<li><p>Create a double-entered file by calling the
’CreatePairLinksDoubleEntered` function. At minimum, pass the (i)
outcome dataset, the (ii) linking dataset, and the (iii) name(s) of the
outcome variable(s). <em>(There are occasions when a single-entered file
is more appropriate for a DF analysis. See Rodgers &amp; Kohler, 2005,
for additional information.)</em></p></li>
<li><p>Use ’DeFriesFulkerMethod3` function (<em>i.e.</em>, general
linear model) to estimate the coefficients of the DF model.</p></li>
</ol>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="do">### R Code for Example DF analysis with a simple outcome and Gen2 subjects</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co"># Step 2: Load the package containing the linking routines.</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="fu">library</span>(NlsyLinks)</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co"># Step 3: Load the LINKING dataset and filter for the Gen2 subjects</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>dsLinking <span class="ot">&lt;-</span> <span class="fu">subset</span>(Links79Pair, RelationshipPath <span class="sc">==</span> <span class="st">&quot;Gen2Siblings&quot;</span>)</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="fu">summary</span>(dsLinking) <span class="co"># Notice there are 11,088 records (one for each unique pair).</span></span></code></pre></div>
<pre><code>##    ExtendedID    SubjectTag_S1     SubjectTag_S2           R        
##  Min.   :    2   Min.   :    201   Min.   :    202   Min.   :0.250  
##  1st Qu.: 3154   1st Qu.: 315401   1st Qu.: 315403   1st Qu.:0.250  
##  Median : 6105   Median : 610703   Median : 610705   Median :0.500  
##  Mean   : 5927   Mean   : 593061   Mean   : 593063   Mean   :0.417  
##  3rd Qu.: 8507   3rd Qu.: 851001   3rd Qu.: 851003   3rd Qu.:0.500  
##  Max.   :12673   Max.   :1267301   Max.   :1267302   Max.   :1.000  
##        RelationshipPath
##  Gen1Housemates:    0  
##  Gen2Siblings  :11114  
##  Gen2Cousins   :    0  
##  ParentChild   :    0  
##  AuntNiece     :    0  
## </code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co"># Step 4: Load the OUTCOMES dataset, and then examine the summary.</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>dsOutcomes <span class="ot">&lt;-</span> ExtraOutcomes79 <span class="co">#&#39; ds&#39; stands for &#39;Data Set&#39;</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="fu">summary</span>(dsOutcomes)</span></code></pre></div>
<pre><code>##    SubjectTag        SubjectID         Generation    HeightZGenderAge WeightZGenderAge
##  Min.   :    100   Min.   :      1   Min.   :1.000   Min.   :-2.985   Min.   :-2.985  
##  1st Qu.: 314025   1st Qu.:   5998   1st Qu.:1.000   1st Qu.:-0.724   1st Qu.:-0.677  
##  Median : 620050   Median :  12000   Median :1.000   Median :-0.045   Median :-0.149  
##  Mean   : 618600   Mean   : 289254   Mean   :1.476   Mean   :-0.006   Mean   : 0.001  
##  3rd Qu.: 914501   3rd Qu.: 577403   3rd Qu.:2.000   3rd Qu.: 0.648   3rd Qu.: 0.533  
##  Max.   :1268600   Max.   :1267501   Max.   :2.000   Max.   : 2.996   Max.   : 4.945  
##                                                      NA&#39;s   :4711     NA&#39;s   :4719    
##  AfqtRescaled2006Gaussified      Afi             Afm        MathStandardized
##  Min.   :-2.895             Min.   : 2.00   Min.   : 0.00   Min.   : 65.0   
##  1st Qu.:-0.692             1st Qu.:15.00   1st Qu.:12.00   1st Qu.: 92.5   
##  Median :-0.024             Median :17.00   Median :13.00   Median :100.0   
##  Mean   :-0.011             Mean   :16.66   Mean   :12.78   Mean   :100.1   
##  3rd Qu.: 0.660             3rd Qu.:18.00   3rd Qu.:14.00   3rd Qu.:108.5   
##  Max.   : 2.994             Max.   :27.00   Max.   :19.00   Max.   :135.0   
##  NA&#39;s   :12510              NA&#39;s   :12740   NA&#39;s   :18165   NA&#39;s   :15085</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co"># Step 5: This step isn&#39;t necessary for this example, because Kelly Meredith already</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co">#   groomed the values.  If the negative values (which represent NLSY missing or</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co">#   skip patterns) still exist, then:</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>dsOutcomes<span class="sc">$</span>MathStandardized[dsOutcomes<span class="sc">$</span>MathStandardized <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co"># Step 6: Create the double entered dataset.</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>dsDouble <span class="ot">&lt;-</span> <span class="fu">CreatePairLinksDoubleEntered</span>(</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>  <span class="at">outcomeDataset   =</span> dsOutcomes,</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>  <span class="at">linksPairDataset =</span> dsLinking,</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>  <span class="at">outcomeNames     =</span> <span class="fu">c</span>(<span class="st">&quot;MathStandardized&quot;</span>)</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>)</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="fu">summary</span>(dsDouble) <span class="co"># Notice there are 22176=(2*11088) records now (two for each unique pair).</span></span></code></pre></div>
<pre><code>##  SubjectTag_S1     SubjectTag_S2       ExtendedID          R        
##  Min.   :    201   Min.   :    201   Min.   :    2   Min.   :0.250  
##  1st Qu.: 315402   1st Qu.: 315402   1st Qu.: 3154   1st Qu.:0.250  
##  Median : 610704   Median : 610704   Median : 6105   Median :0.500  
##  Mean   : 593062   Mean   : 593062   Mean   : 5927   Mean   :0.417  
##  3rd Qu.: 851002   3rd Qu.: 851002   3rd Qu.: 8508   3rd Qu.:0.500  
##  Max.   :1267302   Max.   :1267302   Max.   :12673   Max.   :1.000  
##                                                                     
##        RelationshipPath MathStandardized_S1 MathStandardized_S2
##  Gen1Housemates:    0   Min.   : 65.0       Min.   : 65.0      
##  Gen2Siblings  :22228   1st Qu.: 90.0       1st Qu.: 90.0      
##  Gen2Cousins   :    0   Median : 98.5       Median : 98.5      
##  ParentChild   :    0   Mean   : 98.4       Mean   : 98.4      
##  AuntNiece     :    0   3rd Qu.:107.0       3rd Qu.:107.0      
##                         Max.   :135.0       Max.   :135.0      
##                         NA&#39;s   :3926        NA&#39;s   :3926</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co"># Step 7: Estimate the ACE components with a DF Analysis</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>ace <span class="ot">&lt;-</span> <span class="fu">DeFriesFulkerMethod3</span>(</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>  <span class="at">dataSet  =</span> dsDouble,</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>  <span class="at">oName_S1 =</span> <span class="st">&quot;MathStandardized_S1&quot;</span>,</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>  <span class="at">oName_S2 =</span> <span class="st">&quot;MathStandardized_S2&quot;</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>)</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>ace</span></code></pre></div>
<pre><code>## [1] &quot;Results of ACE estimation: [show]&quot;
##     ASquared     CSquared     ESquared    CaseCount 
## 7.734447e-01 1.469204e-01 7.963486e-02 1.668000e+04</code></pre>
<p><em>Further Information</em>: If the different reasons of missingness
are important, further work is necessary. For instance, some analyses
that use item <code>Y19940000</code> might need to distinguish a
response of “Don’t Know” (which is coded as -2) from “Missing” (which is
coded as -7). For this vignette example, we’ll assume it’s safe to clump
the responses together.</p>
</div>
<div id="example-df-analysis-with-a-univariate-outcome-from-a-gen2-extract" class="section level1">
<h1>Example: DF analysis with a univariate outcome from a Gen2
Extract</h1>
<p>The vignette’s second example differs from the previous example in
two ways. First, the outcome variables are read from a CSV (<a href="https://en.wikipedia.org/wiki/Comma-separated_values">comma
separated values</a> file) that was downloaded from the <strong>NLS
Investigator</strong>. Second, the DF analysis is called through the
function <code>AceUnivariate</code>; this function is a wrapper around
some simple ACE methods, and will help us smoothly transition to more
techniques later in the vignette.</p>
<p>The steps are:</p>
<ol style="list-style-type: decimal">
<li><p>Use the NLS Investigator to select and download a Gen2 dataset.
Select the variables ‘length of gestation of child in weeks’
(<code>C03280.00</code>), ‘weight of child at birth in ounces’
(<code>C03286.00</code>), and ‘length of child at birth’
(<code>C03288.00</code>), and then download the *.zip file to your local
computer.</p></li>
<li><p><a href="#appendix-creating-and-saving-r-scripts">Open R and
create a new script</a> and load the NlsyLinks package.</p></li>
<li><p>Within the R script, load the linking dataset. Then select only
Gen2 subjects.</p></li>
<li><p>Read the CSV into R as a <code>data.frame</code> using
<code>ReadCsvNlsy79Gen2</code>.</p></li>
<li><p>Verify the desired outcome column exists, and rename it something
meaningful to your project. It is important that the
<code>data.frame</code> is reassigned (<em>i.e.</em>,
<code>ds &lt;- RenameNlsyColumn(...)</code>). In this example, we rename
column <code>C0328800</code> to
<code>BirthWeightInOunces</code>.</p></li>
<li><p>Filter out all subjects who have a negative
<code>BirthWeightInOunces</code> value. See the ‘Further Information’
note in the previous example.</p></li>
<li><p>Create a double-entered file by calling the
<code>CreatePairLinksDoubleEntered</code> function. At minimum, pass the
(i) outcome dataset, the (ii) linking dataset, and the (iii) name(s) of
the outcome variable(s).</p></li>
<li><p>Call the <code>AceUnivariate</code> function to estimate the
coefficients.</p></li>
</ol>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="do">### R Code for Example of a DF analysis with a simple outcome and Gen2 subjects</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="co"># Step 2: Load the package containing the linking routines.</span></span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="fu">library</span>(NlsyLinks)</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="co"># Step 3: Load the linking dataset and filter for the Gen2 subjects</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>dsLinking <span class="ot">&lt;-</span> <span class="fu">subset</span>(Links79Pair, RelationshipPath <span class="sc">==</span> <span class="st">&quot;Gen2Siblings&quot;</span>)</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="co"># Step 4: Load the outcomes dataset from the hard drive and then examine the summary.</span></span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a><span class="co">#   Your path might be: filePathOutcomes &lt;- &#39;C:/BGResearch/NlsExtracts/gen2-birth.csv&#39;</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>filePathOutcomes <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">path.package</span>(<span class="st">&quot;NlsyLinks&quot;</span>), <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;gen2-birth.csv&quot;</span>)</span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>dsOutcomes <span class="ot">&lt;-</span> <span class="fu">ReadCsvNlsy79Gen2</span>(filePathOutcomes)</span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a><span class="fu">summary</span>(dsOutcomes)</span></code></pre></div>
<pre><code>##    SubjectTag        SubjectID         ExtendedID      Generation SubjectTagOfMother
##  Min.   :    201   Min.   :    201   Min.   :    2   Min.   :2    Min.   :    200   
##  1st Qu.: 310302   1st Qu.: 310302   1st Qu.: 3101   1st Qu.:2    1st Qu.: 310300   
##  Median : 604607   Median : 604607   Median : 6045   Median :2    Median : 604600   
##  Mean   : 601313   Mean   : 601313   Mean   : 6007   Mean   :2    Mean   : 601311   
##  3rd Qu.: 876203   3rd Qu.: 876203   3rd Qu.: 8757   3rd Qu.:2    3rd Qu.: 876200   
##  Max.   :1267501   Max.   :1267501   Max.   :12675   Max.   :2    Max.   :1267500   
##                                      NA&#39;s   :2                                      
##     C0005300        C0005400         C0005700       C0328000        C0328600    
##  Min.   :1.000   Min.   :-3.000   Min.   :  -3   Min.   :-7.00   Min.   : -7.0  
##  1st Qu.:2.000   1st Qu.: 1.000   1st Qu.:1981   1st Qu.:37.00   1st Qu.: 99.0  
##  Median :3.000   Median : 1.000   Median :1985   Median :39.00   Median :115.0  
##  Mean   :2.338   Mean   : 1.489   Mean   :1986   Mean   :33.51   Mean   :103.9  
##  3rd Qu.:3.000   3rd Qu.: 2.000   3rd Qu.:1990   3rd Qu.:39.00   3rd Qu.:128.0  
##  Max.   :3.000   Max.   : 2.000   Max.   :2008   Max.   :51.00   Max.   :768.0  
##                                                                                 
##     C0328800    
##  Min.   :-7.00  
##  1st Qu.:18.00  
##  Median :20.00  
##  Mean   :16.51  
##  3rd Qu.:21.00  
##  Max.   :48.00  
## </code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co"># Step 5: Verify and rename an existing column.</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="fu">VerifyColumnExists</span>(dsOutcomes, <span class="st">&quot;C0328600&quot;</span>) <span class="co"># Should return &#39;10&#39; in this example.</span></span></code></pre></div>
<pre><code>## [1] 10</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>dsOutcomes <span class="ot">&lt;-</span> <span class="fu">RenameNlsyColumn</span>(dsOutcomes, <span class="st">&quot;C0328600&quot;</span>, <span class="st">&quot;BirthWeightInOunces&quot;</span>)</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a><span class="co"># Step 6: For this item, a negative value indicates the parent refused, didn&#39;t know,</span></span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a><span class="co">#   invalidly skipped, or was missing for some other reason.</span></span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a><span class="co">#   For our present purposes, we&#39;ll treat these responses equivalently.</span></span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a><span class="co">#   Then clip/Winsorized/truncate the weight to something reasonable.</span></span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>dsOutcomes<span class="sc">$</span>BirthWeightInOunces[dsOutcomes<span class="sc">$</span>BirthWeightInOunces <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>dsOutcomes<span class="sc">$</span>BirthWeightInOunces <span class="ot">&lt;-</span> <span class="fu">pmin</span>(dsOutcomes<span class="sc">$</span>BirthWeightInOunces, <span class="dv">200</span>)</span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a><span class="co"># Step 7: Create the double entered dataset.</span></span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a>dsDouble <span class="ot">&lt;-</span> <span class="fu">CreatePairLinksDoubleEntered</span>(</span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>  <span class="at">outcomeDataset   =</span> dsOutcomes,</span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>  <span class="at">linksPairDataset =</span> dsLinking,</span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a>  <span class="at">outcomeNames     =</span> <span class="fu">c</span>(<span class="st">&quot;BirthWeightInOunces&quot;</span>)</span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a>)</span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a></span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a><span class="co"># Step 8: Estimate the ACE components with a DF Analysis</span></span>
<span id="cb16-18"><a href="#cb16-18" tabindex="-1"></a>ace <span class="ot">&lt;-</span> <span class="fu">AceUnivariate</span>(</span>
<span id="cb16-19"><a href="#cb16-19" tabindex="-1"></a>  <span class="at">method   =</span> <span class="st">&quot;DeFriesFulkerMethod3&quot;</span>,</span>
<span id="cb16-20"><a href="#cb16-20" tabindex="-1"></a>  <span class="at">dataSet  =</span> dsDouble,</span>
<span id="cb16-21"><a href="#cb16-21" tabindex="-1"></a>  <span class="at">oName_S1 =</span> <span class="st">&quot;BirthWeightInOunces_S1&quot;</span>,</span>
<span id="cb16-22"><a href="#cb16-22" tabindex="-1"></a>  <span class="at">oName_S2 =</span> <span class="st">&quot;BirthWeightInOunces_S2&quot;</span></span>
<span id="cb16-23"><a href="#cb16-23" tabindex="-1"></a>)</span>
<span id="cb16-24"><a href="#cb16-24" tabindex="-1"></a>ace</span></code></pre></div>
<pre><code>## [1] &quot;Results of ACE estimation: [show]&quot;
##     ASquared     CSquared     ESquared    CaseCount 
## 5.103108e-01 1.752415e-01 3.144477e-01 1.744000e+04</code></pre>
<p>For another example of incorporating CSVs downloaded from the NLS
Investigator, please see the “Race and Gender Variables” entry in the <a href="https://nlsy-links.github.io/NlsyLinks/articles/faq.html">FAQ</a>.</p>
</div>
<div id="example-multiple-group-sem-of-a-simple-outcome-for-gen2-subjects" class="section level1">
<h1>Example: Multiple Group SEM of a Simple Outcome for Gen2
Subjects</h1>
<p>The example differs from the first one by the statistical mechanism
used to estimate the components. The first example uses multiple
regression to estimate the influence of the shared genetic and
environmental factors, while this example uses structural equation
modeling (SEM).</p>
<p>The <code>CreatePairLinksSingleEntered</code> function will create a
<code>data.frame</code> where each row represents one unique pair of
siblings, <em>irrespective of order</em>. Other than producing half the
number of rows, this function is identical to
<code>CreatePairLinksDoubleEntered</code>.</p>
<p>The steps are:</p>
<p>(Steps 1-5 proceed identically to the first example.)</p>
<ol start="6" style="list-style-type: decimal">
<li><p>Create a <em>single</em>-entered file by calling the
<code>CreatePairLinksSingleEntered</code> function. At minimum, pass the
(i) outcome dataset, the (ii) linking dataset, and the (iii) name(s) of
the outcome variable(s).</p></li>
<li><p>Declare the names of the outcome variables corresponding to the
two members in each pair. Assuming the variable is called ‘ZZZ’ and the
preceding steps have been followed, the variable ‘ZZZ_S1’ corresponds to
the first members and ZZZ_S2’ corresponds to the second
members.</p></li>
<li><p>Create a GroupSummary <code>data.frame</code>, which identifies
the <code>R</code> groups that should be considered by the model.
Inspect the output to see if the groups show unexpected or fishy
differences.</p></li>
<li><p>Create a <code>data.frame</code> with cleaned variables to pass
to the SEM function. This <code>data.frame</code> contains only the
three necessary rows and columns.</p></li>
<li><p>Estimate the SEM with the package. The function returns an
<code>S4</code> object, which shows the basic ACE information.</p></li>
<li><p>Inspect details of the SEM, beyond the ACE components. In this
example, we look at the fit stats and the parameter estimates. The
package has additional methods that may be useful for your
purposes.</p></li>
</ol>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="do">### R Code for Example lavaan estimation analysis with a simple outcome and Gen2 subjects</span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="co"># Steps 1-5 are explained in the vignette&#39;s first example:</span></span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a><span class="fu">library</span>(NlsyLinks)</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>dsLinking <span class="ot">&lt;-</span> <span class="fu">subset</span>(Links79Pair, RelationshipPath <span class="sc">==</span> <span class="st">&quot;Gen2Siblings&quot;</span>)</span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>dsOutcomes <span class="ot">&lt;-</span> ExtraOutcomes79</span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>dsOutcomes<span class="sc">$</span>MathStandardized[dsOutcomes<span class="sc">$</span>MathStandardized <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a><span class="co"># Step 6: Create the single entered dataset.</span></span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a>dsSingle <span class="ot">&lt;-</span> <span class="fu">CreatePairLinksSingleEntered</span>(</span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a>  <span class="at">outcomeDataset   =</span> dsOutcomes,</span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a>  <span class="at">linksPairDataset =</span> dsLinking,</span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a>  <span class="at">outcomeNames     =</span> <span class="fu">c</span>(<span class="st">&quot;MathStandardized&quot;</span>)</span>
<span id="cb18-13"><a href="#cb18-13" tabindex="-1"></a>)</span>
<span id="cb18-14"><a href="#cb18-14" tabindex="-1"></a></span>
<span id="cb18-15"><a href="#cb18-15" tabindex="-1"></a><span class="co"># Step 7: Declare the names for the two outcome variables.</span></span>
<span id="cb18-16"><a href="#cb18-16" tabindex="-1"></a>oName_S1 <span class="ot">&lt;-</span> <span class="st">&quot;MathStandardized_S1&quot;</span> <span class="co"># Stands for Outcome1</span></span>
<span id="cb18-17"><a href="#cb18-17" tabindex="-1"></a>oName_S2 <span class="ot">&lt;-</span> <span class="st">&quot;MathStandardized_S2&quot;</span> <span class="co"># Stands for Outcome2</span></span>
<span id="cb18-18"><a href="#cb18-18" tabindex="-1"></a></span>
<span id="cb18-19"><a href="#cb18-19" tabindex="-1"></a><span class="co"># Step 8: Summarize the R groups and determine which groups can be estimated.</span></span>
<span id="cb18-20"><a href="#cb18-20" tabindex="-1"></a>dsGroupSummary <span class="ot">&lt;-</span> <span class="fu">RGroupSummary</span>(dsSingle, oName_S1, oName_S2)</span>
<span id="cb18-21"><a href="#cb18-21" tabindex="-1"></a>dsGroupSummary</span></code></pre></div>
<pre><code>##       R Included PairCount    O1Mean    O2Mean O1Variance O2Variance O1O2Covariance
## 1 0.250     TRUE      2689  95.10450  95.97936   126.9489   150.1775       41.96914
## 2 0.375     TRUE       137  93.63139  93.36861   160.0120   136.6628       50.39790
## 3 0.500     TRUE      5491  99.89374 100.02868   168.7326   172.7293       90.04116
## 4 0.750    FALSE         2 108.50000 106.00000   220.5000    18.0000       63.00000
## 5 1.000     TRUE        21  98.21429  96.02381   289.4393   215.2369      229.10714
##   Correlation Determinant PosDefinite
## 1   0.3039577   17303.459        TRUE
## 2   0.3408090   19327.735        TRUE
## 3   0.5274225   21037.642        TRUE
## 4   1.0000000       0.000       FALSE
## 5   0.9179130    9807.933        TRUE</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="co"># Step 9: Create a cleaned dataset</span></span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>dsClean <span class="ot">&lt;-</span> <span class="fu">CleanSemAceDataset</span>(<span class="at">dsDirty =</span> dsSingle, dsGroupSummary, oName_S1, oName_S2)</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a><span class="co"># Step 10: Run the model</span></span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>ace <span class="ot">&lt;-</span> <span class="fu">AceLavaanGroup</span>(dsClean)</span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a>ace</span></code></pre></div>
<pre><code>## [1] &quot;Results of ACE estimation: [show]&quot;
##     ASquared     CSquared     ESquared    CaseCount 
##    0.6219254    0.2097338    0.1683407 8338.0000000</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="co"># Notice the &#39;CaseCount&#39; is 8,390 instead of 17,440.</span></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a><span class="co">#  This is because (a) one pair with R=.75 was excluded, and</span></span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a><span class="co">#  (b) the SEM uses a single-entered dataset instead of double-entered.</span></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a><span class="co"># Step 11: Inspect the output further</span></span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a><span class="fu">library</span>(lavaan) <span class="co"># Load the package to access methods of the lavaan class.</span></span></code></pre></div>
<pre><code>## This is lavaan 0.6-19
## lavaan is FREE software! Please report any bugs.</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="fu">GetDetails</span>(ace)</span></code></pre></div>
<pre><code>## lavaan 0.6-19 ended normally after 39 iterations
## 
##   Estimator                                         ML
##   Optimization method                           NLMINB
##   Number of model parameters                        32
##   Number of equality constraints                    28
## 
##   Number of observations per group:                   
##     1                                             2689
##     2                                              137
##     3                                             5491
##     4                                               21
## 
## Model Test User Model:
##                                                       
##   Test statistic                               447.241
##   Degrees of freedom                                16
##   P-value (Chi-square)                           0.000
##   Test statistic for each group:
##     1                                          281.866
##     2                                           30.277
##     3                                          127.671
##     4                                            7.428</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="co"># Examine fit stats like Chi-Squared, RMSEA, CFI, etc.</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a><span class="fu">fitMeasures</span>(<span class="fu">GetDetails</span>(ace)) <span class="co">#&#39; fitMeasures&#39; is defined in the lavaan package.</span></span></code></pre></div>
<pre><code>##                  npar                  fmin                 chisq                    df 
##                 4.000                 0.027               447.241                16.000 
##                pvalue        baseline.chisq           baseline.df       baseline.pvalue 
##                 0.000              2106.324                 4.000                 0.000 
##                   cfi                   tli                  nnfi                   rfi 
##                 0.795                 0.949                 0.949                    NA 
##                   nfi                  pnfi                   ifi                   rni 
##                    NA                 3.151                 0.794                 0.795 
##                  logl     unrestricted.logl                   aic                   bic 
##            -65103.779            -64880.158            130215.557            130243.671 
##                ntotal                  bic2                 rmsea        rmsea.ci.lower 
##              8338.000            130230.960                 0.114                 0.105 
##        rmsea.ci.upper        rmsea.ci.level          rmsea.pvalue        rmsea.close.h0 
##                 0.123                 0.900                 0.000                 0.050 
## rmsea.notclose.pvalue     rmsea.notclose.h0                   rmr            rmr_nomean 
##                 1.000                 0.080                 9.992                12.765 
##                  srmr          srmr_bentler   srmr_bentler_nomean                  crmr 
##                 0.130                 0.130                 0.089                 0.138 
##           crmr_nomean            srmr_mplus     srmr_mplus_nomean                 cn_05 
##                 0.026                 0.153                 0.083               491.246 
##                 cn_01                   gfi                  agfi                  pgfi 
##               597.581                 0.999                 0.999                 0.799 
##                   mfi                  ecvi 
##                 0.974                 0.055</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="co"># Examine low-level details like each group&#39;s individual parameter estimates and standard</span></span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a><span class="co">#  errors.  Uncomment the next line to view the entire output (which is roughly 4 pages).</span></span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a><span class="co"># summary(GetDetails(ace))</span></span></code></pre></div>
</div>
<div id="example-multiple-group-sem-of-a-simple-outcome-for-gen1-subjects" class="section level1">
<h1>Example: Multiple Group SEM of a Simple Outcome for Gen1
Subjects</h1>
<p>The example differs from the previous one in three ways. First, Gen1
subjects are used. Second, standardized height is used instead of math.
Third, pairs are dropped if their <code>R</code> is zero; we return to
this last issue after the code is run.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="do">### R Code for Example lavaan estimation analysis with a simple outcome and Gen1 subjects</span></span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a><span class="co"># Steps 1-5 are explained in the vignette&#39;s first example:</span></span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a><span class="fu">library</span>(NlsyLinks)</span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a>dsLinking <span class="ot">&lt;-</span> <span class="fu">subset</span>(Links79Pair, RelationshipPath <span class="sc">==</span> <span class="st">&quot;Gen1Housemates&quot;</span>)</span>
<span id="cb29-5"><a href="#cb29-5" tabindex="-1"></a>dsOutcomes <span class="ot">&lt;-</span> ExtraOutcomes79</span>
<span id="cb29-6"><a href="#cb29-6" tabindex="-1"></a><span class="co"># The HeightZGenderAge variable is already groomed</span></span>
<span id="cb29-7"><a href="#cb29-7" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" tabindex="-1"></a><span class="co"># Step 6: Create the single entered dataset.</span></span>
<span id="cb29-9"><a href="#cb29-9" tabindex="-1"></a>dsSingle <span class="ot">&lt;-</span> <span class="fu">CreatePairLinksSingleEntered</span>(</span>
<span id="cb29-10"><a href="#cb29-10" tabindex="-1"></a>  <span class="at">outcomeDataset   =</span> dsOutcomes,</span>
<span id="cb29-11"><a href="#cb29-11" tabindex="-1"></a>  <span class="at">linksPairDataset =</span> dsLinking,</span>
<span id="cb29-12"><a href="#cb29-12" tabindex="-1"></a>  <span class="at">outcomeNames     =</span> <span class="fu">c</span>(<span class="st">&quot;HeightZGenderAge&quot;</span>)</span>
<span id="cb29-13"><a href="#cb29-13" tabindex="-1"></a>)</span>
<span id="cb29-14"><a href="#cb29-14" tabindex="-1"></a></span>
<span id="cb29-15"><a href="#cb29-15" tabindex="-1"></a><span class="co"># Step 7: Declare the names for the two outcome variables.</span></span>
<span id="cb29-16"><a href="#cb29-16" tabindex="-1"></a>oName_S1 <span class="ot">&lt;-</span> <span class="st">&quot;HeightZGenderAge_S1&quot;</span></span>
<span id="cb29-17"><a href="#cb29-17" tabindex="-1"></a>oName_S2 <span class="ot">&lt;-</span> <span class="st">&quot;HeightZGenderAge_S2&quot;</span></span>
<span id="cb29-18"><a href="#cb29-18" tabindex="-1"></a></span>
<span id="cb29-19"><a href="#cb29-19" tabindex="-1"></a><span class="co"># Step 8: Summarize the R groups and determine which groups can be estimated.</span></span>
<span id="cb29-20"><a href="#cb29-20" tabindex="-1"></a>dsGroupSummary <span class="ot">&lt;-</span> <span class="fu">RGroupSummary</span>(dsSingle, oName_S1, oName_S2)</span>
<span id="cb29-21"><a href="#cb29-21" tabindex="-1"></a>dsGroupSummary</span></code></pre></div>
<pre><code>##      R Included PairCount      O1Mean       O2Mean O1Variance O2Variance O1O2Covariance
## 1 0.25     TRUE       280  0.04830041  0.056049671  1.0181579  1.1847214      0.2655632
## 2 0.50     TRUE      3894 -0.04986899 -0.027888465  0.9735617  1.0193789      0.4660286
## 3 1.00     TRUE        11 -0.08652006 -0.009039755  0.3171247  0.9518269      0.3583306
##   Correlation Determinant PosDefinite
## 1   0.2417977   1.1357096        TRUE
## 2   0.4678030   0.7752456        TRUE
## 3   0.6522137   0.1734469        TRUE</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="co"># Step 9: Create a cleaned dataset</span></span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>dsClean <span class="ot">&lt;-</span> <span class="fu">CleanSemAceDataset</span>(<span class="at">dsDirty =</span> dsSingle, dsGroupSummary, oName_S1, oName_S2)</span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a><span class="co"># Step 10: Run the model</span></span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a>ace <span class="ot">&lt;-</span> <span class="fu">AceLavaanGroup</span>(dsClean)</span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a>ace</span></code></pre></div>
<pre><code>## [1] &quot;Results of ACE estimation: [show]&quot;
##     ASquared     CSquared     ESquared    CaseCount 
##    0.7040149    0.1123658    0.1836193 4185.0000000</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="co"># Step 11: Inspect the output further (see the final step in the previous example).</span></span></code></pre></div>
<p>Most of them responded they were <code>Non-relative</code>s to the
explicit items asked in 1979 (<em>i.e.</em>, NLSY79 variables
<code>R00001.50</code> through <code>R00001.59</code>). Yet their
height’s observed correlations is far larger than would be expected for
a sample of unrelated subjects. Since our team began BG research with
the NLSY in the mid-1990s, the <span class="math inline">\(R\)</span>=0
group has consistently presented higher than expected correlations,
across many domains of outcome variables. For a long time, we have
substantial doubts that subject pairs in this group share a low
proportion of their selective genes. Consequently, we suggest applied
researchers consider excluding this group from their biometric
analyses.</p>
<p>If you wish to exclude additional groups from the analyses, Step 8
should change slightly. For instance, to MZ twins, replace the two lines
in Step 8 with the following four. This is most for demonstration. It is
unlikely to be useful idea in the current example, and is more likely to
be useful when using the <code>RFull</code> variable, which includes all
values of <code>R</code> we were able to determine.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="co"># Step 8: Summarize the R groups and determine which groups can be estimated.</span></span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>dsGroupSummary <span class="ot">&lt;-</span> <span class="fu">RGroupSummary</span>(dsSingle, oName_S1, oName_S2)</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a>rGroupsToDrop <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>)</span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>dsGroupSummary[dsGroupSummary<span class="sc">$</span>R <span class="sc">%in%</span> rGroupsToDrop, <span class="st">&quot;Included&quot;</span>] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a>dsGroupSummary</span></code></pre></div>
<pre><code>##      R Included PairCount      O1Mean       O2Mean O1Variance O2Variance O1O2Covariance
## 1 0.25     TRUE       280  0.04830041  0.056049671  1.0181579  1.1847214      0.2655632
## 2 0.50     TRUE      3894 -0.04986899 -0.027888465  0.9735617  1.0193789      0.4660286
## 3 1.00    FALSE        11 -0.08652006 -0.009039755  0.3171247  0.9518269      0.3583306
##   Correlation Determinant PosDefinite
## 1   0.2417977   1.1357096        TRUE
## 2   0.4678030   0.7752456        TRUE
## 3   0.6522137   0.1734469        TRUE</code></pre>
</div>
<div id="example-multiple-group-sem-of-a-simple-outcome-all-pairs-in-gen1-and-gen2" class="section level1">
<h1>Example: Multiple Group SEM of a Simple Outcome all pairs in Gen1
and Gen2</h1>
<p>The example differs from the previous example in one way –all
possible pairs are considered for the analysis. Pairs are only excluded
(a) if they belong to one of the small <code>R</code> groups that are
difficult to estimate, or (b) if the value for adult height is missing.
This includes all relationships in the follow five types of NLSY79
relationships.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>xt <span class="ot">&lt;-</span> <span class="fu">xtable</span>(<span class="fu">table</span>(Links79Pair<span class="sc">$</span>RelationshipPath, <span class="at">dnn =</span> <span class="fu">c</span>(<span class="st">&quot;Relationship Frequency&quot;</span>)),</span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>  <span class="at">caption =</span> <span class="st">&quot;Number of NLSY79 relationship, by `RelationshipPath`.(Recall that &#39;AuntNiece&#39; also contains uncles and nephews.)&quot;</span></span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a>)</span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a><span class="fu">print.xtable</span>(xt, <span class="at">format.args =</span> <span class="fu">list</span>(<span class="at">big.mark =</span> <span class="st">&quot;,&quot;</span>), <span class="at">type =</span> <span class="st">&quot;html&quot;</span>)</span></code></pre></div>
<!-- html table generated in R 4.4.1 by xtable 1.8-4 package -->
<!-- Mon Oct  7 15:13:45 2024 -->
<table border="1">
<caption align="bottom">
Number of NLSY79 relationship, by <code>RelationshipPath</code>.(Recall
that ‘AuntNiece’ also contains uncles and nephews.)
</caption>
<tr>
<th>
</th>
<th>
Relationship Frequency
</th>
</tr>
<tr>
<td align="right">
Gen1Housemates
</td>
<td align="right">
5,302
</td>
</tr>
<tr>
<td align="right">
Gen2Siblings
</td>
<td align="right">
11,114
</td>
</tr>
<tr>
<td align="right">
Gen2Cousins
</td>
<td align="right">
5,000
</td>
</tr>
<tr>
<td align="right">
ParentChild
</td>
<td align="right">
11,521
</td>
</tr>
<tr>
<td align="right">
AuntNiece
</td>
<td align="right">
9,899
</td>
</tr>
</table>
<p>In our opinion, using the intergenerational links is one of the most
exciting new opportunities for NLSY researchers to pursue. We will be
happy to facilitate such research through consult or collaboration, or
even by generating new data structures that may be of value. The
complete kinship linking file facilitates many different kinds of
cross-generational research, using both biometrical and other kinds of
modeling methods.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="do">### R Code for Example lavaan estimation analysis with a simple outcome and Gen1 subjects</span></span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a><span class="co"># Steps 1-5 are explained in the vignette&#39;s first example:</span></span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a><span class="fu">library</span>(NlsyLinks)</span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a>dsLinking <span class="ot">&lt;-</span> <span class="fu">subset</span>(Links79Pair, RelationshipPath <span class="sc">%in%</span></span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a>  <span class="fu">c</span>(</span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a>    <span class="st">&quot;Gen1Housemates&quot;</span>, <span class="st">&quot;Gen2Siblings&quot;</span>, <span class="st">&quot;Gen2Cousins&quot;</span>,</span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a>    <span class="st">&quot;ParentChild&quot;</span>, <span class="st">&quot;AuntNiece&quot;</span></span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a>  ))</span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a><span class="co"># Because all five paths are specified, the line above is equivalent to:</span></span>
<span id="cb37-10"><a href="#cb37-10" tabindex="-1"></a><span class="co"># dsLinking &lt;- Links79Pair</span></span>
<span id="cb37-11"><a href="#cb37-11" tabindex="-1"></a></span>
<span id="cb37-12"><a href="#cb37-12" tabindex="-1"></a>dsOutcomes <span class="ot">&lt;-</span> ExtraOutcomes79</span>
<span id="cb37-13"><a href="#cb37-13" tabindex="-1"></a><span class="co"># The HeightZGenderAge variable is already groomed</span></span>
<span id="cb37-14"><a href="#cb37-14" tabindex="-1"></a></span>
<span id="cb37-15"><a href="#cb37-15" tabindex="-1"></a><span class="co"># Step 6: Create the single entered dataset.</span></span>
<span id="cb37-16"><a href="#cb37-16" tabindex="-1"></a>dsSingle <span class="ot">&lt;-</span> <span class="fu">CreatePairLinksSingleEntered</span>(</span>
<span id="cb37-17"><a href="#cb37-17" tabindex="-1"></a>  <span class="at">outcomeDataset   =</span> dsOutcomes,</span>
<span id="cb37-18"><a href="#cb37-18" tabindex="-1"></a>  <span class="at">linksPairDataset =</span> dsLinking,</span>
<span id="cb37-19"><a href="#cb37-19" tabindex="-1"></a>  <span class="at">outcomeNames     =</span> <span class="fu">c</span>(<span class="st">&quot;HeightZGenderAge&quot;</span>)</span>
<span id="cb37-20"><a href="#cb37-20" tabindex="-1"></a>)</span>
<span id="cb37-21"><a href="#cb37-21" tabindex="-1"></a></span>
<span id="cb37-22"><a href="#cb37-22" tabindex="-1"></a><span class="co"># Step 7: Declare the names for the two outcome variables.</span></span>
<span id="cb37-23"><a href="#cb37-23" tabindex="-1"></a>oName_S1 <span class="ot">&lt;-</span> <span class="st">&quot;HeightZGenderAge_S1&quot;</span></span>
<span id="cb37-24"><a href="#cb37-24" tabindex="-1"></a>oName_S2 <span class="ot">&lt;-</span> <span class="st">&quot;HeightZGenderAge_S2&quot;</span></span>
<span id="cb37-25"><a href="#cb37-25" tabindex="-1"></a></span>
<span id="cb37-26"><a href="#cb37-26" tabindex="-1"></a><span class="co"># Step 8: Summarize the R groups and determine which groups can be estimated.</span></span>
<span id="cb37-27"><a href="#cb37-27" tabindex="-1"></a>dsGroupSummary <span class="ot">&lt;-</span> <span class="fu">RGroupSummary</span>(dsSingle, oName_S1, oName_S2)</span>
<span id="cb37-28"><a href="#cb37-28" tabindex="-1"></a>dsGroupSummary</span></code></pre></div>
<pre><code>##        R Included PairCount      O1Mean      O2Mean O1Variance O2Variance
## 1 0.0625     TRUE       202  0.22968753 -0.07575395  1.0509023  0.8271487
## 2 0.1250     TRUE      2422 -0.02213186  0.01502572  1.0334694  0.9809845
## 3 0.2500     TRUE      7136 -0.05441460 -0.03885252  1.0189426  1.0294526
## 4 0.3750     TRUE        48  0.13536238 -0.10786196  1.1497643  0.9269109
## 5 0.5000     TRUE     14862 -0.05738494 -0.01738686  0.9602559  0.9851045
## 6 0.7500    FALSE         0          NA          NA         NA         NA
## 7 1.0000     TRUE        27 -0.11092113 -0.12601206  0.6934418  0.9891146
##   O1O2Covariance Correlation Determinant PosDefinite
## 1      0.1112158   0.1192871   0.8568836        TRUE
## 2      0.1487539   0.1477368   0.9916897        TRUE
## 3      0.2721042   0.2656790   0.9749124        TRUE
## 4      0.4074779   0.3947123   0.8996908        TRUE
## 5      0.4146816   0.4263636   0.7739915        TRUE
## 6             NA          NA          NA       FALSE
## 7      0.6863894   0.8287857   0.2147630        TRUE</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="co"># Step 9: Create a cleaned dataset</span></span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a>dsClean <span class="ot">&lt;-</span> <span class="fu">CleanSemAceDataset</span>(<span class="at">dsDirty =</span> dsSingle, dsGroupSummary, oName_S1, oName_S2)</span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a></span>
<span id="cb39-4"><a href="#cb39-4" tabindex="-1"></a><span class="co"># Step 10: Run the model</span></span>
<span id="cb39-5"><a href="#cb39-5" tabindex="-1"></a>ace <span class="ot">&lt;-</span> <span class="fu">AceLavaanGroup</span>(dsClean)</span>
<span id="cb39-6"><a href="#cb39-6" tabindex="-1"></a>ace</span></code></pre></div>
<pre><code>## [1] &quot;Results of ACE estimation: [show]&quot;
##     ASquared     CSquared     ESquared    CaseCount 
## 7.366543e-01 6.580003e-02 1.975457e-01 2.469700e+04</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="co"># Step 11: Inspect the output further (see the final step two examples above).</span></span></code></pre></div>
<p>Notice the ACE estimates are very similar to the previous version,
but the number of pairs has increased by 6x –from 4,185 to 24,700. The
number of <em>subjects</em> doubles when Gen2 is added, and the number
of <em>relationship pairs</em> really takes off. When an extended
family’s entire pedigree is considered by the model, many more types of
links are possible than if just nuclear families are considered. This
increased statistical power is even more important when the population’s
<span class="math inline">\(a^2\)</span> is small or moderate, instead
of something large like 0.7.</p>
<p>You may notice that the analysis has 24,697 relationships instead of
the entire 42,836. This is primarily because not all subjects have a
value for ‘adult height’ (and that’s mostly because a lot of Gen2
subjects are too young). There are 42,088 pairs with a nonmissing value
in <code>RFull</code>, meaning that 98.3 are classified. We feel
comfortable claiming that if a researcher has a phenotype for both
members of a pair, there’s a 99+% chance we have an <code>RFull</code>
for it. For a description of the <code>R</code> and <code>RFull</code>
variables, please see the <code>Links79Pair</code> entry in the package
<a href="https://nlsy-links.github.io/NlsyLinks/">reference
manual</a>.</p>
<p><strong>References:</strong></p>
<p>The standard errors (but not the coefficients) are biased downward in
these analyses, because individuals are included in multiple pairs. Our
MDAN article presents a <a href="https://en.wikipedia.org/wiki/Generalized_estimating_equation">GEE</a>
method for handling this (p. 572). The CARB model (or any model that
treats the full pedigree as a single unit of analysis in the
multivariate or multilevel sense) also would produce more accurate
standard error estimates.</p>
<p>One of our <a href="https://r-forge.r-project.org/forum/forum.php?thread_id=28498&amp;forum_id=4266&amp;group_id=1330">2013
BGA presentations</a> discusses these benefits in the context of the
current NlsyLinks package, and our 2008 MDAN article accomplishes
something similar using a GEE with females in both generations.</p>
<p>Bard, D.E., Beasley, W.H., Meredith, K., &amp; Rodgers, J.L. (2012).
<a href="https://link.springer.com/article/10.1007/s10519-012-9566-6"><em>Biometric
Analysis of Complex NLSY Pedigrees: Introducing a Conditional
Autoregressive Biometric (CARB) Mixed Model</em></a>. Behavior Genetics
Association 42nd Annual Meeting. [<a href="https://r-forge.r-project.org/forum/forum.php?thread_id=4761&amp;forum_id=4266&amp;group_id=1330">Slides</a>]</p>
<p>Beasley, W.H., Bard, D.E., Meredith, K., Hunter, M., &amp; Rodgers,
J.L. (2013). <a href="https://link.springer.com/article/10.1007/s10519-013-9623-9"><em>NLSY
Kinship Links: Creating Biometrical Design Structures from
Cross-Generational Data</em></a>. Behavior Genetics Association 43rd
Annual Meeting. [<a href="https://r-forge.r-project.org/forum/forum.php?thread_id=28498&amp;forum_id=4266&amp;group_id=1330">Slides</a>]</p>
<p>Rodgers, J. L., Bard, D., Johnson, A., D’Onofrio, B., &amp; Miller,
W. B. (2008). <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2712575/">The
Cross-Generational Mother-Daughter-Aunt-Niece Design: Establishing
Validity of the MDAN Design with NLSY Fertility Variables</a>.
<em>Behavior Genetics, 38</em>, 567-578.</p>
</div>
<div id="appendix-receiving-help-for-the-nlsylinks-package" class="section level1">
<h1>Appendix: Receiving Help for the NlsyLinks Package</h1>
<p>A portion of our current grant covers a small, part-time support
staff. If you have questions about BG research with our kinship links,
or questions about our package, we’d like to hear from you.</p>
<p>We provide personal support for researchers in several ways. Perhaps
the best place to start are the forums on R-Forge (<a href="https://r-forge.r-project.org/forum/?group_id=1330" class="uri">https://r-forge.r-project.org/forum/?group_id=1330</a>);
there are forums for people using R, as well as other software such as
SAS. <a href="https://r-forge.r-project.org/forum/forum.php?thread_id=4537&amp;forum_id=4266&amp;group_id=1330">This
post</a> is a good overview of the current project is, which originally
was an email Joe sent to previous users of our kinship links (many of
them are/were SAS users).</p>
</div>
<div id="appendix-creating-and-saving-r-scripts" class="section level1">
<h1>Appendix: Creating and Saving R Scripts</h1>
<p>There are several options and environments for executing R code. Our
current recommendation is <a href="https://posit.co/">RStudio</a>,
because it is easy to install, and has features targeting beginner and
experienced R users. We’ve had good experiences with it on Windows, OS
X, and Ubuntu Linux.</p>
<p>RStudio allows you to create and save R files; these are simply text
files that have an file extension of ‘.R’. RStudio will execute the
commands written in the file.</p>
</div>
<div id="appendix-installing-and-loading-the-nlsylinks-package" class="section level1">
<h1>Appendix: Installing and Loading the NlsyLinks Package</h1>
<p>There are three operations you’ll typically do with a package: (a)
install, (b) load, and (c) update.</p>
<p>The simplest way to <em>install</em> NlsyLinks is to type
<code>install.packages(&quot;NlsyLinks&quot;)</code> in the console.</p>
<p>R then will download NlsyLinks on your local computer. It may try to
save and install the package to a location that you don’t have
permission to write files in. If so, R will ask if you would like to
install it to a better location (<em>i.e.</em>, somewhere you do have
permission to write files). Approve this decision (which is acceptable
for everyone except for some network administrators).</p>
<p>For a given computer, you’ll need to <em>install</em> a package only
once for each version of R (new versions of R are released every few
months). However, you’ll need to <em>load</em> a package in every
session that you call its functions. To <em>load</em> NlsyLinks, type
<code>library(NlsyLinks)</code>. Loading reads NlsyLinks information
from the hard drive and places it in temporary memory. Once it’s loaded,
you won’t need to load it again until R is closed and reopened
later.</p>
<p>Developers are continually improving their packages by adding
functions and documentation. These newer versions are then uploaded to
the CRAN servers. You may <em>update</em> all your installed packages at
once by typing <code>update.packages(ask = FALSE)</code>. The command
checks a CRAN server for newer versions of the packages installed on
your local machine. Then they are automatically downloaded and
installed.</p>
<p>The grant supporting NlsyLinks extends until 2020. Until then, we’ll
be including new features and documentation, as we address additional
user needs (if you have suggestions, we’d like to hear from you). When
the NLSY periodically updates its data, we’ll update our kinship links
(embedded in NlsyLinks) with the newest information.</p>
</div>
<div id="appendix-references" class="section level1">
<h1>Appendix: References</h1>
<p>A list of some articles that have used the NLSY for behavior genetics
is available at: <a href="https://nlsy-links.github.io/NlsyLinks/articles/publications.html" class="uri">https://nlsy-links.github.io/NlsyLinks/articles/publications.html</a>.</p>
<p>Carey, Gregory (2002). <a href="https://www.colorado.edu/psych-neuro/gregory-carey"><em>Human
Genetics for the Social Sciences</em></a>. Sage.</p>
<p>Plomin, Robert (1990). <a href="https://books.google.com/books?id=r7AgAQAAIAAJ&amp;source=gbs_navlinks_s"><em>Nature
and nurture: an introduction to human behavioral genetics</em></a>.
Brooks/Cole Publishing Company.</p>
<p>Rodgers, J. L., Bard, D., Johnson, A., D’Onofrio, B., &amp; Miller,
W. B. (2008). <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2712575/">The
Cross-Generational Mother-Daughter-Aunt-Niece Design: Establishing
Validity of the MDAN Design with NLSY Fertility Variables</a>.
<em>Behavior Genetics, 38</em>, 567-578.</p>
<p>Rodgers, Joseph Lee, &amp; Kohler, Hans-Peter (2005). <a href="https://pubmed.ncbi.nlm.nih.gov/15685433/">Reformulating and
simplifying the DF analysis model</a>. <em>Behavior Genetics, 35
(2)</em>, 211-217.</p>
<p>Rodgers, Joseph Lee, Rowe, David C., &amp; Li, Chengchang (1994). <a href="https://psycnet.apa.org:443/journals/dev/30/3/374/">Beyond nature
versus nurture: DF analysis of nonshared influences on problem
behaviors</a>. <em>Developmental Psychology, 30 (3)</em>, 374-384.</p>
<p>Neale, Michael C., &amp; Cardon, Lou R. (1992). <a href="https://books.google.com/books/about/Methodology_for_genetic_studies_of_twins.html?id=vVzDmDv6WDkC"><em>Methodology
for genetic studies of twins and families</em></a>. Norwell, MA: Kluwer
Academic Publishers. (Also see Neale &amp; Maes: [<a href="http://ibgwww.colorado.edu/workshop2006/cdrom/HTML/book2004a.pdf" class="uri">http://ibgwww.colorado.edu/workshop2006/cdrom/HTML/book2004a.pdf</a>]).</p>
</div>
<div id="funding" class="section level1">
<h1>Funding</h1>
<p>This package’s development has been supported by two grants from NIH.
The first, NIH Grant <a href="https://taggs.hhs.gov/Detail/AwardDetail?arg_awardNum=R01HD065865&amp;arg_ProgOfficeCode=50">1R01HD65865</a>,
“NLSY Kinship Links: Reliable and Valid Sibling Identiﬁcation” (PI: Joe
Rodgers; Vignette Construction by Will Beasley) supported the
(virtually) final completion of the NLSY79 and NLSYC/YA kinship linking
files. The second, NIH Grant <a href="https://reporter.nih.gov/project-details/9239744">1R01HD087395</a>,
“New NLSY Kinship Links and Longitudinal/ Cross-Generational Models:
Cognition and Fertility Research,” (PI: Joe Rodgers; Vignette
Construction by Will Beasley) is supporting the development of the
NLSY97 kinship links, and slight updates/extensions in the links for the
two earlier data sources.</p>
</div>
<div id="version-information" class="section level1">
<h1>Version Information</h1>
<p>For the sake of documentation and reproducibility, the current report
was rendered in the following environment. Click the line below to
expand. But you’ll probably regret it.</p>
<details>
<summary>
Environment <span class="glyphicon glyphicon-plus-sign"></span>
</summary>
<pre><code>##  setting  value
##  version  R version 4.4.1 (2024-06-14 ucrt)
##  os       Windows 11 x64 (build 22631)
##  system   x86_64, mingw32
##  ui       RTerm
##  language (EN)
##  collate  C
##  ctype    English_United States.utf8
##  tz       America/New_York
##  date     2024-10-07
##  pandoc   3.2 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
loadedversion
</th>
<th style="text-align:left;">
date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
bslib
</td>
<td style="text-align:left;">
0.8.0
</td>
<td style="text-align:left;">
2024-07-29
</td>
</tr>
<tr>
<td style="text-align:left;">
cachem
</td>
<td style="text-align:left;">
1.1.0
</td>
<td style="text-align:left;">
2024-05-16
</td>
</tr>
<tr>
<td style="text-align:left;">
cli
</td>
<td style="text-align:left;">
3.6.3
</td>
<td style="text-align:left;">
2024-06-21
</td>
</tr>
<tr>
<td style="text-align:left;">
colorspace
</td>
<td style="text-align:left;">
2.1-1
</td>
<td style="text-align:left;">
2024-07-26
</td>
</tr>
<tr>
<td style="text-align:left;">
devtools
</td>
<td style="text-align:left;">
2.4.5
</td>
<td style="text-align:left;">
2022-10-11
</td>
</tr>
<tr>
<td style="text-align:left;">
digest
</td>
<td style="text-align:left;">
0.6.37
</td>
<td style="text-align:left;">
2024-08-19
</td>
</tr>
<tr>
<td style="text-align:left;">
ellipsis
</td>
<td style="text-align:left;">
0.3.2
</td>
<td style="text-align:left;">
2021-04-29
</td>
</tr>
<tr>
<td style="text-align:left;">
evaluate
</td>
<td style="text-align:left;">
1.0.0
</td>
<td style="text-align:left;">
2024-09-17
</td>
</tr>
<tr>
<td style="text-align:left;">
fastmap
</td>
<td style="text-align:left;">
1.2.0
</td>
<td style="text-align:left;">
2024-05-15
</td>
</tr>
<tr>
<td style="text-align:left;">
fs
</td>
<td style="text-align:left;">
1.6.4
</td>
<td style="text-align:left;">
2024-04-25
</td>
</tr>
<tr>
<td style="text-align:left;">
glue
</td>
<td style="text-align:left;">
1.7.0
</td>
<td style="text-align:left;">
2024-01-09
</td>
</tr>
<tr>
<td style="text-align:left;">
htmltools
</td>
<td style="text-align:left;">
0.5.8.1
</td>
<td style="text-align:left;">
2024-04-04
</td>
</tr>
<tr>
<td style="text-align:left;">
htmlwidgets
</td>
<td style="text-align:left;">
1.6.4
</td>
<td style="text-align:left;">
2023-12-06
</td>
</tr>
<tr>
<td style="text-align:left;">
httpuv
</td>
<td style="text-align:left;">
1.6.15
</td>
<td style="text-align:left;">
2024-03-26
</td>
</tr>
<tr>
<td style="text-align:left;">
jquerylib
</td>
<td style="text-align:left;">
0.1.4
</td>
<td style="text-align:left;">
2021-04-26
</td>
</tr>
<tr>
<td style="text-align:left;">
jsonlite
</td>
<td style="text-align:left;">
1.8.9
</td>
<td style="text-align:left;">
2024-09-20
</td>
</tr>
<tr>
<td style="text-align:left;">
knitr
</td>
<td style="text-align:left;">
1.48
</td>
<td style="text-align:left;">
2024-07-07
</td>
</tr>
<tr>
<td style="text-align:left;">
later
</td>
<td style="text-align:left;">
1.3.2
</td>
<td style="text-align:left;">
2023-12-06
</td>
</tr>
<tr>
<td style="text-align:left;">
lavaan
</td>
<td style="text-align:left;">
0.6-19
</td>
<td style="text-align:left;">
2024-09-26
</td>
</tr>
<tr>
<td style="text-align:left;">
lifecycle
</td>
<td style="text-align:left;">
1.0.4
</td>
<td style="text-align:left;">
2023-11-07
</td>
</tr>
<tr>
<td style="text-align:left;">
magrittr
</td>
<td style="text-align:left;">
2.0.3
</td>
<td style="text-align:left;">
2022-03-30
</td>
</tr>
<tr>
<td style="text-align:left;">
MASS
</td>
<td style="text-align:left;">
7.3-61
</td>
<td style="text-align:left;">
2024-06-13
</td>
</tr>
<tr>
<td style="text-align:left;">
memoise
</td>
<td style="text-align:left;">
2.0.1
</td>
<td style="text-align:left;">
2021-11-26
</td>
</tr>
<tr>
<td style="text-align:left;">
mime
</td>
<td style="text-align:left;">
0.12
</td>
<td style="text-align:left;">
2021-09-28
</td>
</tr>
<tr>
<td style="text-align:left;">
miniUI
</td>
<td style="text-align:left;">
0.1.1.1
</td>
<td style="text-align:left;">
2018-05-18
</td>
</tr>
<tr>
<td style="text-align:left;">
mnormt
</td>
<td style="text-align:left;">
2.1.1
</td>
<td style="text-align:left;">
2022-09-26
</td>
</tr>
<tr>
<td style="text-align:left;">
munsell
</td>
<td style="text-align:left;">
0.5.1
</td>
<td style="text-align:left;">
2024-04-01
</td>
</tr>
<tr>
<td style="text-align:left;">
NlsyLinks
</td>
<td style="text-align:left;">
2.2.2
</td>
<td style="text-align:left;">
2024-10-07
</td>
</tr>
<tr>
<td style="text-align:left;">
pbivnorm
</td>
<td style="text-align:left;">
0.6.0
</td>
<td style="text-align:left;">
2015-01-23
</td>
</tr>
<tr>
<td style="text-align:left;">
pkgbuild
</td>
<td style="text-align:left;">
1.4.4
</td>
<td style="text-align:left;">
2024-03-17
</td>
</tr>
<tr>
<td style="text-align:left;">
pkgload
</td>
<td style="text-align:left;">
1.4.0
</td>
<td style="text-align:left;">
2024-06-28
</td>
</tr>
<tr>
<td style="text-align:left;">
profvis
</td>
<td style="text-align:left;">
0.4.0
</td>
<td style="text-align:left;">
2024-09-20
</td>
</tr>
<tr>
<td style="text-align:left;">
promises
</td>
<td style="text-align:left;">
1.3.0
</td>
<td style="text-align:left;">
2024-04-05
</td>
</tr>
<tr>
<td style="text-align:left;">
purrr
</td>
<td style="text-align:left;">
1.0.2
</td>
<td style="text-align:left;">
2023-08-10
</td>
</tr>
<tr>
<td style="text-align:left;">
quadprog
</td>
<td style="text-align:left;">
1.5-8
</td>
<td style="text-align:left;">
2019-11-20
</td>
</tr>
<tr>
<td style="text-align:left;">
R6
</td>
<td style="text-align:left;">
2.5.1
</td>
<td style="text-align:left;">
2021-08-19
</td>
</tr>
<tr>
<td style="text-align:left;">
Rcpp
</td>
<td style="text-align:left;">
1.0.13
</td>
<td style="text-align:left;">
2024-07-17
</td>
</tr>
<tr>
<td style="text-align:left;">
remotes
</td>
<td style="text-align:left;">
2.5.0
</td>
<td style="text-align:left;">
2024-03-17
</td>
</tr>
<tr>
<td style="text-align:left;">
rlang
</td>
<td style="text-align:left;">
1.1.4
</td>
<td style="text-align:left;">
2024-06-04
</td>
</tr>
<tr>
<td style="text-align:left;">
rmarkdown
</td>
<td style="text-align:left;">
2.28
</td>
<td style="text-align:left;">
2024-08-17
</td>
</tr>
<tr>
<td style="text-align:left;">
rstudioapi
</td>
<td style="text-align:left;">
0.16.0
</td>
<td style="text-align:left;">
2024-03-24
</td>
</tr>
<tr>
<td style="text-align:left;">
sass
</td>
<td style="text-align:left;">
0.4.9
</td>
<td style="text-align:left;">
2024-03-15
</td>
</tr>
<tr>
<td style="text-align:left;">
scales
</td>
<td style="text-align:left;">
1.3.0
</td>
<td style="text-align:left;">
2023-11-28
</td>
</tr>
<tr>
<td style="text-align:left;">
sessioninfo
</td>
<td style="text-align:left;">
1.2.2
</td>
<td style="text-align:left;">
2021-12-06
</td>
</tr>
<tr>
<td style="text-align:left;">
shiny
</td>
<td style="text-align:left;">
1.9.1
</td>
<td style="text-align:left;">
2024-08-01
</td>
</tr>
<tr>
<td style="text-align:left;">
urlchecker
</td>
<td style="text-align:left;">
1.0.1
</td>
<td style="text-align:left;">
2021-11-30
</td>
</tr>
<tr>
<td style="text-align:left;">
usethis
</td>
<td style="text-align:left;">
3.0.0
</td>
<td style="text-align:left;">
2024-07-29
</td>
</tr>
<tr>
<td style="text-align:left;">
vctrs
</td>
<td style="text-align:left;">
0.6.5
</td>
<td style="text-align:left;">
2023-12-01
</td>
</tr>
<tr>
<td style="text-align:left;">
xfun
</td>
<td style="text-align:left;">
0.47
</td>
<td style="text-align:left;">
2024-08-17
</td>
</tr>
<tr>
<td style="text-align:left;">
xtable
</td>
<td style="text-align:left;">
1.8-4
</td>
<td style="text-align:left;">
2019-04-21
</td>
</tr>
<tr>
<td style="text-align:left;">
yaml
</td>
<td style="text-align:left;">
2.3.10
</td>
<td style="text-align:left;">
2024-07-26
</td>
</tr>
</tbody>
</table>
</details>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
